python/lsst/ap/association/diaPipe.py

```diff
@@ -585,72 +761,59 @@ def run(self,
-                self.log.warning(
-                    "Duplicate DiaForcedSources created after merge with "
-                    "history and new sources. This may cause downstream "
-                    "problems. Dropping duplicates.")
-                # Drop duplicates via index and keep the first appearance.
-                # Reset due to the index shape being slight different than
-                # expected.
-                diaForcedSources = diaForcedSources.groupby(
-                    diaForcedSources.index).first()
-                diaForcedSources.reset_index(drop=True, inplace=True)
-                diaForcedSources.set_index(
-                    ["diaObjectId", "diaForcedSourceId"],
-                    drop=False,
-                    inplace=True)
-            self.alertPackager.run(associatedDiaSources,
-                                   diaCalResult.diaObjectCat,
-                                   preloadedDiaSources,
-                                   diaForcedSources,
-                                   diffIm,
-                                   exposure,
-                                   template,
-                                   doRunForcedMeasurement=self.config.doRunForcedMeasurement,
-                                   )
-
-        # For historical reasons, apdbMarker is a Config even if it's not meant to be read.
-        # A default Config is the cheapest way to satisfy the storage class.
-        marker = self.config.apdb.value if self.config.doConfigureApdb else pexConfig.Config()
-        return pipeBase.Struct(apdbMarker=marker,
-                               associatedDiaSources=associatedDiaSources,
-                               diaForcedSources=diaForcedSources,
-                               diaObjects=diaObjects,
-                               )
-
-    def createNewDiaObjects(self, unAssocDiaSources):
-        """Loop through the set of DiaSources and create new DiaObjects
-        for unassociated DiaSources.
+    @timeMethod
+    def packageAlerts(self, diffIm, exposure, template, preloadedDiaSources, preloadedDiaForcedSources,
+                      associatedDiaSources, diaForcedSources, diaObjects):
+        """Package alerts.
 
         Parameters
         ----------
-        unAssocDiaSources : `pandas.DataFrame`
-            Set of DiaSources to create new DiaObjects from.
-
-        Returns
-        -------
-        results : `lsst.pipe.base.Struct`
-            Results struct containing:
-
-            - ``diaSources`` : DiaSource catalog with updated DiaObject ids.
-              (`pandas.DataFrame`)
-            - ``newDiaObjects`` : Newly created DiaObjects from the
-              unassociated DiaSources. (`pandas.DataFrame`)
-            - ``nNewDiaObjects`` : Number of newly created diaObjects.(`int`)
+        diffIm : `lsst.afw.image.ExposureF`
+            Difference image exposure in which the sources in ``diaSourceCat``
```

Comment:
The reference in back-ticks needs to match the parameter name. With my suggestion, that would mean `diaSourceCat` should be `diaSources`